### PR TITLE
fix: address PR #302 review feedback, update spec submodule

### PR DESCRIPTION
**Pull request by amis92**
Addresses review feedback from PR #302, updates spec submodule, and fixes edge cases.

### Null/empty ID edge cases (review threads 1-3)
- `JoinLinkPrefix` now treats empty `id` as a no-op — prevents dangling `"::"` segments
- `BuildEntryIdPath` refactored to use `JoinLinkPrefix` for link path construction, matching `ComputeCompositeEntryId` exactly
- `entryGroupId` passes `null` when `sourceGroup.Id` is null/empty

### Selection ordering stability (review thread 5)
- Switched from unstable `ImmutableArray.Sort` to stable LINQ `OrderBy`
- Removed `EntryId` tiebreaker — equal-name selections preserve FIFO insertion order

### Other improvements
- `GetForceCategoryIds` return type changed from `HashSet<string>?` to `HashSet<string>` (never returns null)

### Spec submodule update
- Updated `lib/battlescribe-spec` to latest `main` (aa18776) which includes fix for contradictory ordering defaults (WarHub/battlescribe-spec#220)
- wham no longer needs any engine-specific overrides

All 353 conformance tests pass.

### CODE REVIEW

File: src/WarHub.ArmouryModel.Concrete.Extensions/SelectionOrdering.cs [hunk 1]
```diff
@@ -70,9 +70,12 @@ public int Compare(ISelectionSymbol? a, ISelectionSymbol? b)
...
                 || b.EffectiveSourceEntry.Name != b.SourceEntry?.Name)
             {
-                return NaturalSort.Compare(a.SourceEntry?.Name ?? a.Name, b.SourceEntry?.Name ?? b.Name);
+                cmp = NaturalSort.Compare(a.SourceEntry?.Name ?? a.Name, b.SourceEntry?.Name ?? b.Name);
+                if (cmp != 0) return cmp;
             }
-            return 0;
+            // Stable tiebreaker: compare by EntryId to ensure deterministic ordering
+            // when effective names match (ImmutableArray.Sort is unstable).
+            return string.Compare(a.EntryId, b.EntryId, StringComparison.Ordinal);
```
[the remark this answers is not in the document]
Comment (amis92) [reply src/WarHub.ArmouryModel.Concrete.Extensions/SelectionOrdering.cs]: The ordering should default to the selection order (fifo, last added is last on list)

File: src/WarHub.ArmouryModel.RosterEngine/EntryResolver.cs [hunk 2]
```diff
@@ -116,7 +116,7 @@ public static IReadOnlyList<AvailableEntry> GetChildEntries(
 
         // If this entry is a link, children accumulate its ID into the prefix.
         var childPrefix = entry.ReferencedEntry is not null
-            ? JoinLinkPrefix(contextLinkPrefix, entry.Id!)
+            ? JoinLinkPrefix(contextLinkPrefix, entry.Id ?? "")
             : contextLinkPrefix;
```
[the remark this answers is not in the document]
Comment (amis92) [reply src/WarHub.ArmouryModel.RosterEngine/EntryResolver.cs:121]: Fixed in 1ad306d — `JoinLinkPrefix` now treats empty `id` as a no-op (returns prefix unchanged), preventing dangling `"::"` segments. The same fix applies to all three call sites (lines 121, 173, 222).

File: src/WarHub.ArmouryModel.RosterEngine/WhamRosterEngine.cs [hunk 3]
```diff
@@ -446,7 +446,7 @@ private static SelectionNode CreateSelectionNode(
         var entryId = BuildEntryIdPath(entry, linkPrefix);
         var entryGroupId = sourceGroup is null
             ? null
-            : EntryResolver.JoinLinkPrefix(linkPrefix, sourceGroup.Id!);
+            : EntryResolver.JoinLinkPrefix(linkPrefix, sourceGroup.Id ?? "");
 
         var selectionNode = NodeFactory.Selection(entryDecl, entryId, entryGroupId);
```
[the remark this answers is not in the document]
Comment (amis92) [reply src/WarHub.ArmouryModel.RosterEngine/WhamRosterEngine.cs:451]: Fixed in 1ad306d — `entryGroupId` now passes `null` when `sourceGroup.Id` is null/empty (using `sourceGroup?.Id is { Length: > 0 }` guard), avoiding empty string or malformed prefix serialization.

File: src/WarHub.ArmouryModel.RosterEngine/WhamRosterEngine.cs [hunk 4]
```diff
@@ -1192,11 +1192,15 @@ public MutationResult SelectChildEntryById(
...
+        // use the composite form (e.g., "linkId::targetId") while the input entryId
+        // may be in raw form.
+        var compositeEntryId = EntryResolver.ComputeCompositeEntryId(childAvail);
         if (IsNumberIncrementEntry(childAvail.Symbol))
         {
             foreach (var existingChild in parentSel.Selections)
             {
-                if (existingChild.EntryId == entryId)
+                if (existingChild.EntryId == compositeEntryId)
                 {
```
[the remark this answers is not in the document]
Comment (amis92) [reply src/WarHub.ArmouryModel.RosterEngine/WhamRosterEngine.cs:1200]: Fixed in 1ad306d — `BuildEntryIdPath` now uses `JoinLinkPrefix(JoinLinkPrefix(prefix, linkId), targetId)` for links, matching `ComputeCompositeEntryId` exactly. Combined with `JoinLinkPrefix` treating empty `id` as a no-op, both methods now produce identical results for null/empty link IDs.

[the remark this answers is not in the document]
Comment (amis92) [reply src/WarHub.ArmouryModel.RosterEngine/WhamRosterEngine.cs:1200]: This behavior is exercised by the conformance test suite via `entry-id-*` specs which test various link prefix / composite ID scenarios end-to-end. Adding a dedicated unit test for the duplicate detection path is reasonable future work but not blocking for this PR.

File: src/WarHub.ArmouryModel.Concrete.Extensions/SelectionOrdering.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (amis92) [reply src/WarHub.ArmouryModel.Concrete.Extensions/SelectionOrdering.cs]: Fixed in 1ad306d — per @amis92''s feedback, switched to stable sort (LINQ `OrderBy`) and removed the `EntryId` tiebreaker. Equal-name selections now preserve their insertion order (FIFO). `ImmutableArray.Sort` was unstable; `OrderBy` guarantees stability.

[the remark this answers is not in the document]
Comment (amis92) [reply src/WarHub.ArmouryModel.Concrete.Extensions/SelectionOrdering.cs]: Selection ordering is covered by the conformance test suite (e.g., `scope-child-id-filter`, `scope-primary-catalogue` specs verify expected selection order). The stable sort fix reduces the risk of regressions for equal-name elements. Dedicated ordering unit tests are reasonable future work.
